src/main/java/fr/openent/competences/service/impl/DefaultNoteService.java

```diff
@@ -3508,109 +3516,123 @@ public void getDataGraph(final String idEleve, JsonArray groupIds, final String
         competenceNoteService.getConversionNoteCompetence(idEtablissement, idClasse, tableauEvent ->
                 formate(tableauDeConversionFuture, tableauEvent));
 
-        CompositeFuture.all(compNoteF, noteF, subjectF, tableauDeConversionFuture).setHandler(event -> {
-            if(event.failed()){
-                String message = "[getReleveDataForGraph] " + event.cause().getMessage();
-                log.error(message);
-                handler.handle(new Either.Left<>(message));
-                return;
-            }
+        Promise<Boolean> isAvgSkillpromise = Promise.promise() ;
```

Comment:
Tu pourrais dans DefaultStructureOptions ajouter la méthode `isAverageSkills` qui reprendrait la méthode pour seulement retourner true comme on s'aperçoit qu'il est très souvent utilisé : 

DefaultStructureOptions : 
```
@Override
    public Future<Boolean> isAverageSkills(String structureId) {
        Promise<Boolean> promise = Promise.promise();
        getIsAverageSkills(structureId, event -> {
            if(event.isRight())
                promise.complete(event.right().getValue().getBoolean(Field.IS_AVERAGE_SKILLS)); // pour "is_average_skills"
            else
                promise.fail(event.left().getValue());
        });
        return promise.future();
    }
```

Et donc par la suite faire : 

```suggestion
        Future<Boolean> isAvgSkillpromise = structureOptionsService.isAverageSkills(idEtablissement);
                listFutures.add(isAvgSkillpromise);
```